src/main/java/seedu/address/model/appointment/Appointment.java

```diff
@@ -0,0 +1,112 @@
+package seedu.address.model.appointment;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.person.Nric;
+
+/**
+ * Appointment class that describes an appointment
+ */
+public class Appointment {
+
+    private static final String MESSAGE_CONSTRAINTS_INVALID_DATE =
+            "Appointment should be made with a date today onwards";
+
+    // The doctor in charge of the appointment
+    private final Nric doctorNric;
+
+    // The patient benefiting from the appointment
+    private final Nric patientNric;
+
+    // The date of the appointment
+    private final AppointmentDate appointmentDate;
+
+    // Message to outputs in case constraints are not met
+
+    private final AppointmentId appointmentId;
+
+    /**
+     * Constructs a new appointment instance
+     * @param doctor doctor in charge
+     * @param patient patient of the appointment
+     * @param appointmentDate date of the appointment
+     */
+    public Appointment(Nric doctorNric, Nric patientNric, AppointmentDate appointmentDate) {
+        requireAllNonNull(doctorNric, patientNric, appointmentDate);
+        checkArgument(isValidAppointment(appointmentDate), MESSAGE_CONSTRAINTS_INVALID_DATE);
+        this.doctorNric = doctorNric;
+        this.patientNric = patientNric;
+        this.appointmentDate = appointmentDate;
+        this.appointmentId = new AppointmentId();
+    }
+
+    /**
+     * Checks if appointment is valid by comparing appointment date against current date.
+     * A valid new appointment can only be in the future, not the past.
+     *
+     * @param appointmentDate Date to check validity of
+     * @return boolean if appointment is valid or not
+     */
+    private boolean isValidAppointment(AppointmentDate appointmentDate) {
```

Comment:
Would it be more suitable to name the variable 'isValidAppointment' to something like 'isAppointmentDateValid' for future-proofing? E.g In the future when we check whether doctorId or patientId is valid, then 'isValidAppointment' will not just check for date validity.